lib/auth/auth_with_roles.go

```diff
@@ -4557,16 +4566,18 @@ func (a *ServerWithRoles) UpsertRole(ctx context.Context, role types.Role) (type
 		return nil, trace.Wrap(err)
 	}
 
-	verbs := []string{types.VerbCreate}
-	if oldRole != nil {
-		verbs = []string{types.VerbUpdate}
-		if err := a.actionForResource(oldRole, types.KindRole, verbs...); err != nil {
+	switch {
+	case oldRole != nil:
+		if err := a.actionForResource(oldRole, types.KindRole, types.VerbUpdate); err != nil {
```

Comment:
Took me a minute to grok this change. A comment might help.

```suggestion
		// if the role already exists, then require update permission
		// on both the current state of the role and the new state of the role
		if err := a.actionForResource(oldRole, types.KindRole, types.VerbUpdate); err != nil {
```